scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt

```diff
@@ -1042,27 +1045,8 @@ internal class ContainerFactory(
 
     envVars.putAll(getEventHubEnvVars(organization, workspace))
 
-    val template = getRunTemplate(solution, runTemplateId)
-    val csmSimulation = template.csmSimulation
-    if (csmSimulation != null) {
-      envVars[CSM_SIMULATION_VAR] = csmSimulation
-    }
-    val source = step.source?.invoke(template)
-    if (source != null) {
-      envVars[step.providerVar] = source
-      envVars[AZURE_STORAGE_CONNECTION_STRING] =
-          csmPlatformProperties.azure?.storage?.connectionString ?: ""
-      if (organization.id == null) throw IllegalStateException("Organization id cannot be null")
-      if (workspace.id == null) throw IllegalStateException("Workspace id cannot be null")
-      if (source == STEP_SOURCE_CLOUD || source == STEP_SOURCE_PLATFORM) {
-        envVars[step.pathVar] =
-            (step.path?.invoke(organization.id ?: "", solution.id ?: "", runTemplateId) ?: "")
-      } else if (source == STEP_SOURCE_GIT) {
-        envVars[step.pathVar] = template.gitRepositoryUrl ?: ""
-        envVars["CSM_RUN_TEMPLATE_GIT_BRANCH_NAME"] = template.gitBranchName ?: ""
-        envVars["CSM_RUN_TEMPLATE_SOURCE_DIRECTORY"] = template.runTemplateSourceDir ?: ""
-      }
-    }
+    envVars.putAll(getTemplateEnvVars(solution, runTemplateId, step, organization, envVars))
```

Comment:
it will be duplicated map values here: `envVars` from arguments will be added to variable `envVars` with methods `putAll`

Reply:
something like `envVars = getTemplateEnvVars(...) `  ?

Reply:
1. Remove envVars from arguments
2. return a new MutableMap<> constructed in getTemplateEnvVars
3. replace your 2 putAll calls by something like:
`envVars.putAll(getEventHubEnvVars(organization, workspace) + getTemplateEnvVars(solution, runTemplateId, step, organization))`